src/state/index.ts

```diff
@@ -0,0 +1,126 @@
+import axios, { isAxiosError } from 'axios';
+
+import { AdapterState as AdapterStateType, AirdropEvent } from '../types';
+import { STATELESS_EVENT_TYPES } from '../common/constants';
+
+export async function createAdapterState<ExtractorState>(
+  event: AirdropEvent,
+  initialState: ExtractorState
+) {
+  const newInitialState = structuredClone(initialState);
+  const as = new AdapterState<ExtractorState>(event, newInitialState);
+
+  if (!STATELESS_EVENT_TYPES.includes(event.payload.event_type)) {
+    console.log('Fetching state');
+    await as.fetchState(newInitialState);
+  }
+
+  return as;
+}
+
+export class AdapterState<ExtractorState> {
```

Comment:
Please consider renaming this AdapterState class - it has the same naming as the exported AdapterState type.